src/core/src/FabricMaterial.cpp

```diff
@@ -310,18 +303,16 @@ void FabricMaterial::setBaseColorTextureValues(
         scale = {scale.x, scale.y};
     }
 
-    // clang-format off
-    auto textureFabric = srw.getAttributeWr<omni::fabric::AssetPath>(_baseColorTexPathFabric, FabricTokens::inputs_texture);
-    auto texCoordIndexFabric = srw.getAttributeWr<int>(_baseColorTexPathFabric, FabricTokens::inputs_tex_coord_index);
-    auto wrapSFabric = srw.getAttributeWr<int>(_baseColorTexPathFabric, FabricTokens::inputs_wrap_s);
-    auto wrapTFabric = srw.getAttributeWr<int>(_baseColorTexPathFabric, FabricTokens::inputs_wrap_t);
-    auto offsetFabric = srw.getAttributeWr<pxr::GfVec2f>(_baseColorTexPathFabric, FabricTokens::inputs_offset);
-    auto rotationFabric = srw.getAttributeWr<float>(_baseColorTexPathFabric, FabricTokens::inputs_rotation);
-    auto scaleFabric = srw.getAttributeWr<pxr::GfVec2f>(_baseColorTexPathFabric, FabricTokens::inputs_scale);
-    // clang-format on
+    auto textureFabric = srw.getAttributeWr<omni::fabric::AssetPath>(texturePath, FabricTokens::inputs_texture);
+    auto texCoordIndexFabric = srw.getAttributeWr<int>(texturePath, FabricTokens::inputs_tex_coord_index);
+    auto wrapSFabric = srw.getAttributeWr<int>(texturePath, FabricTokens::inputs_wrap_s);
+    auto wrapTFabric = srw.getAttributeWr<int>(texturePath, FabricTokens::inputs_wrap_t);
+    auto offsetFabric = srw.getAttributeWr<pxr::GfVec2f>(texturePath, FabricTokens::inputs_offset);
+    auto rotationFabric = srw.getAttributeWr<float>(texturePath, FabricTokens::inputs_rotation);
+    auto scaleFabric = srw.getAttributeWr<pxr::GfVec2f>(texturePath, FabricTokens::inputs_scale);
 
-    textureFabric->assetPath = pxr::TfToken(textureAssetPath.GetAssetPath());
-    textureFabric->resolvedPath = pxr::TfToken(textureAssetPath.GetResolvedPath());
```

Comment:
In a couple places I was creating these temporary `TfToken`'s which is unsafe since tokens are reference counted and destroyed when the count reaches zero.

Now the texture asset path token is stored in the `FabricTexture` so the token stays around as long as the `FabricTexture` exists.

That's also why the `cesium.mdl` path is passed along as a token instead of an `SdfAssetPath`. The token will exist as long as `Context` exists.